### PR TITLE
plot progress offset index fix

### PR DESCRIPTION
**Pull request by dpendleton22**
added function and flag within utils.py to correct the offset index previously seen in plot_progress, the process_pbt flag in the function is defaulted to False

### CODE REVIEW

File: rlgear/utils.py [hunk 1]
```diff
@@ -262,6 +262,21 @@ def shorten_dfs(_dfs: Sequence[pd.DataFrame], max_step: int = None) -> None:
         _dfs[i] = _df[_df.index <= max_step]  # type: ignore
 
 
+def preprocess_pbt_df(df: pd.DataFrame, x_tag: str) -> pd.DataFrame:
```
Comment (esquires) [inline rlgear/utils.py:265]: I had a bit of difficulty following the logic. Can you add a docstring or comments?

File: rlgear/utils.py [hunk 2]
```diff
@@ -262,6 +262,21 @@ def shorten_dfs(_dfs: Sequence[pd.DataFrame], max_step: int = None) -> None:
...
+    tag_diff = '{}_diff'.format(x_tag)
+    df[tag_diff] = df[x_tag].diff()
+    neg_diff = df.loc[df[tag_diff] < 0].index.to_list()
+    if len(neg_diff) == 0:
+        return df
+    else:
+        for _, val in enumerate(neg_diff):
```
Comment (esquires) [inline rlgear/utils.py:272]: Do we need `enumerate` here? It looks like the index is ignored.

File: rlgear/utils.py [hunk 3]
```diff
@@ -262,6 +262,21 @@ def shorten_dfs(_dfs: Sequence[pd.DataFrame], max_step: int = None) -> None:
...
+            start = val
+            end = len(df)
+            df[x_tag][start:end] += df[tag_diff][start-2] +\
```
Comment (esquires) [inline rlgear/utils.py:275]: can we add a space between the `+` and `\`. Obviously it isn't a big deal; this just matches the rest of the code.

**Review by esquires — COMMENTED** [review]
Thanks! A few small requests.

File: rlgear/utils.py (hunk 1 above)
Comment (dpendleton22) [reply rlgear/utils.py:265]: added a docstring and comments

File: rlgear/utils.py (hunk 2 above)
Comment (dpendleton22) [reply rlgear/utils.py]: removed the enumerate and simply loop through the values in the list

File: rlgear/utils.py (hunk 3 above)
Comment (dpendleton22) [reply rlgear/utils.py]: made this edit

**Comment by esquires** [comment]
Thanks!

**Pull request merged**